packages/editor/src/components/post-taxonomies/flat-term-selector.js

```diff
@@ -56,171 +54,178 @@ const termNamesToIds = ( names, terms ) => {
 	);
 };
 
-class FlatTermSelector extends Component {
-	constructor() {
-		super( ...arguments );
-		this.onChange = this.onChange.bind( this );
-		this.searchTerms = debounce( this.searchTerms.bind( this ), 500 );
-		this.findOrCreateTerm = this.findOrCreateTerm.bind( this );
-		this.appendTerm = this.appendTerm.bind( this );
-		this.state = {
-			loading: ! isEmpty( this.props.terms ),
-			availableTerms: [],
-			selectedTerms: [],
-		};
-	}
+// Tries to create a term or fetch it if it already exists.
+function findOrCreateTerm( termName, restBase ) {
+	const escapedTermName = escapeString( termName );
+
+	return apiFetch( {
+		path: `/wp/v2/${ restBase }`,
+		method: 'POST',
+		data: { name: escapedTermName },
+	} )
+		.catch( ( error ) => {
+			const errorCode = error.code;
+			if ( errorCode === 'term_exists' ) {
```

Comment:
What happens if we don't try to deal with the error at all, is it still necessary?

Reply:
I'm not 100% sure how useful this logic is for actual use-cases. The only way I can trigger an error is if I paster comma-separated  list of tags - `Existing, Existing Tag.`

If a user starts typing and then hits enter, we already have that tag in the suggestions list most of the time. I will do another round of tests with slow network emulated.